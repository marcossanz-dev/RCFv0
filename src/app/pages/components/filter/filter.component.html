<div class="filter-container">
    <ng-content select=".search-form-field"></ng-content>
</div>
