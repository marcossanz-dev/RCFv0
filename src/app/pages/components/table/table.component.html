<ng-template class="table-container" #tableRCF [ngIf]="entities">
    <table mat-table matSort class="mat-elevation-z8" [dataSource]="entities" (matSortChange)="sortData($event)">
        <ng-container *ngFor="let key of columns" [matColumnDef]="key.tag">
            <ng-container *ngIf="key.tag !== 'DGRN'">
                <th mat-header-cell mat-sort-header="{{key.tag}}" *matHeaderCellDef mwlResizable [enableGhostResize]="true" (resizeEnd)="onResizeEnd($event, key.tag)" [resizeEdges]="{bottom: false, right: true, top: false, left: true}">{{key.title}}</th>
                <td mat-cell *matCellDef="let element"> {{element[key.tag]}}</td>
            </ng-container>
            <!-- Checkbox Column -->
            <ng-container *ngIf="key.tag === 'DGRN'">
                <th mat-header-cell mat-sort-header *matHeaderCellDef mwlResizable [enableGhostResize]="true" (resizeEnd)="onResizeEnd($event, key.tag)" [resizeEdges]="{bottom: false, right: true, top: false, left: true}">
                    {{key.title}}
                </th>
                <td mat-cell *matCellDef="let element">
                    <mat-checkbox (click)="$event.stopPropagation()" (change)="$event ?  selection.toggle(element) : null" [checked]="element.DGRN">
                    </mat-checkbox>
                </td>
            </ng-container>
        </ng-container>
        <tr mat-header-row *matHeaderRowDef="displayedColumns "></tr>
        <tr mat-row matRipple id="row-{{index}}" *matRowDef="let row; columns: displayedColumns;let index = index;" (click)="selection.toggle(row)" [ngClass]="{hovered: row.hovered, highlighted: selection.isSelected(row)}" (mouseover)="row.hovered = true" (mouseout)="row.hovered = false">
        </tr>
    </table>
</ng-template>