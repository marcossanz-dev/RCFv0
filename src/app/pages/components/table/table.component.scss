.mat-row:hover {
    background-color: rgb(166, 230, 241);
    cursor: pointer;
}

table {
    width: 100%;
}

mwlResizable {
    box-sizing: border-box;
}

.mat-header-cell,
.mat-cell {
    border-left: 1px solid #eee;
    padding-left: 12px !important;
    padding-right: 12px !important;
    line-height: 46px !important;
}

.mat-row.highlighted {
    background: #999;
}