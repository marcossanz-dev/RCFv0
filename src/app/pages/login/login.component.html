<<<<<<< HEAD
<app-home></app-home>
=======
<h1>Login View</h1>
<div class="login__form">
  <form [formGroup]="loginForm">
    <div class="form-input">
      <label class="label">Username</label>
      <input class="general-input" type="text" formControlName="username" />
      <span class="error-input" *ngIf="this.loginForm.get('username')?.errors"
        >Required</span
      >
    </div>
    <div class="form-input">
      <label class="label">Password</label>
      <input class="general-input" type="password" formControlName="password" />
      <span class="error-input" *ngIf="this.loginForm.get('password')?.errors"
        >Required</span
      >
    </div>
    <div class="form-input">
      <button
        [disabled]="loginForm.invalid"
        (click)="submit()"
        class="submit-button"
      >
        Enter
      </button>
    </div>
  </form>
</div>
>>>>>>> feature/nataliadiego
