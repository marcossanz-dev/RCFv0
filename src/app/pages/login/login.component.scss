form {
  max-width: 31.25rem;
}
.form-input {
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
}

.login__form {
  display: flex;
  justify-content: center;
}

.label {
  font-weight: 500;
  line-height: 3;
  font-size: 16px;
}

.general-input {
  font-size: 14px;
  padding: 12px 5px;
  outline: none;
  background-color: #1976d2;
  color: white;
  border: 0;
  border-radius: 4px;
}

.submit-button {
  font-size: 14px;
  padding: 12px 5px;
  outline: none;
  background-color: #1976d2;
  color: white;
  border: 0;
  border-radius: 4px;
  text-align: center;
  cursor: pointer;
  margin-top: 40px;
  font-weight: 600;

  &:disabled {
    opacity: 0.6;
  }
}

.error-input {
  color: rgb(221, 0, 49);
  font-weight: 500;
  margin-top: 4px;
}
