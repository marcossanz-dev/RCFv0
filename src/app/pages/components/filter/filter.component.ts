import { Component, Input, OnInit } from '@angular/core';
import { MatTableDataSource } from '@angular/material/table';
import { Entity } from 'src/app/libs/entity.interface';

@Component({
  selector: 'app-filter',
  templateUrl: './filter.component.html',
  styleUrls: ['./filter.component.scss']
})
export class FilterComponent implements OnInit {

  @Input() label!: string;
  value!: string;
  constructor() { }

  dataSource: any;

  ngOnInit(): void {

  }

  onChange(el: any) {
    this.value = el.target.value;
    console.log(this.value);
  }

}
