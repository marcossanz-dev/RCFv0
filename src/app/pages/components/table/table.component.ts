import { MatTableDataSource } from '@angular/material/table';
import { SelectionModel } from '@angular/cdk/collections';
import { Component, EventEmitter, OnInit, Output, ViewChild, AfterViewInit } from '@angular/core';
import { MatSort } from '@angular/material/sort';
import { Entity } from 'src/app/libs/entity.interface';
import { Header } from 'src/app/libs/header.interface';
import { ResizeEvent } from 'angular-resizable-element';

const ENTITIES_DATA: Entity[] = [
  { code: 'C0031', DGRN: true, CIF: 'A28013050', name: 'CASER, CAJA DE SEGUROS Y REASEGUROS, S.A', village: 'MADRID', city: 'MADRID', postalCode: '28050', country: 'ESPAÑA', fax: 0 },
  { code: 'C0074', DGRN: false, CIF: 'G08169831', name: 'CAIXA MANRESA', village: 'MANRESA', city: 'BARCELONA', postalCode: '0804', country: 'ESPAÑA', fax: 0 },
  { code: 'C0739', DGRN: false, CIF: 'A82070442', name: 'CAJA ESPAÑA VIDA', village: 'LEON', city: 'LEON', postalCode: '24005', country: 'ESPAÑA', fax: 0 }
];

const HEADER_DATA: Header[] = [
  { tag: 'code', title: 'CODIGO', isChecked: false },
  { tag: 'DGRN', title: 'NOTIFICACION DGRN', isChecked: false },
  { tag: 'CIF', title: 'CIF', isChecked: false },
  { tag: 'name', title: 'DENOMINACIÓN SOCIAL', isChecked: false },
  { tag: 'village', title: 'POBLACIÓN', isChecked: false },
  { tag: 'city', title: 'PROVINCIA', isChecked: false },
  { tag: 'postalCode', title: 'CODIGO POSTAL', isChecked: false },
  { tag: 'country', title: 'PAIS', isChecked: false },
  { tag: 'fax', title: 'FAX', isChecked: false },
  { tag: 'email', title: 'CORREO ELECTRONICO', isChecked: false }
];
@Component({
  selector: 'app-table',
  templateUrl: './table.component.html',
  styleUrls: ['./table.component.scss']
})
export class TableComponent implements AfterViewInit {

  //@Input() data:any[];
  //@Input() displayedColumns: string[];
  @Output() onSelected = new EventEmitter<any>();

  data = new MatTableDataSource(ENTITIES_DATA);
  dataHeader: Header[] = HEADER_DATA;
  displayedColumns: string[];
  columns: any;
  selection = new SelectionModel<Entity>(false, []);

  @ViewChild(MatSort) sort!: MatSort;

  constructor() {
    this.columns = this.dataHeader.map((item) => {
      return { tag: item.tag, title: item.title }
    });
    this.displayedColumns = this.dataHeader.map((item) => {
      return item.tag;
    });
  }


  ngAfterViewInit() {
    this.data.sort = this.sort;
  }

  onResizeEnd(event: ResizeEvent, columnName: any): void {
    if (event.edges.right) {
      const cssValue = event.rectangle.width + 'px';
      const columnElts = document.getElementsByClassName('mat-column-' + columnName);
      for (let i = 0; i < columnElts.length; i++) {
        const currentEl = columnElts[i] as any;
        currentEl.style.width = cssValue;
      }
    }
  }

}