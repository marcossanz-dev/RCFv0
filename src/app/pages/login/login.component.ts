import { Component, OnInit } from '@angular/core';
import {
  FormBuilder,
  FormControl,
  FormGroup,
  Validators,
} from '@angular/forms';
import { Router } from '@angular/router';
import { LoginService } from 'src/app/services/login.service';

@Component({
  selector: 'app-login',
  templateUrl: './login.component.html',
  styleUrls: ['./login.component.scss'],
})
export class LoginComponent {
  loginForm: FormGroup;
  spinner: boolean;

  constructor(
    private loginService: LoginService,
    private router: Router,
    fb: FormBuilder
  ) {
    this.loginForm = fb.group({
      username: ['', Validators.required],
      password: ['', Validators.required],
    });
    this.spinner = false;
  }

  submit(): void {
    if (this.loginForm.valid) {
      this.spinner = true;
      const user = this.loginForm.get('username')?.value;
      const pass = this.loginForm.get('password')?.value;
      this.loginService.login(user, pass).subscribe((res: any) => {
        const jwt = res && res.data ? res.data.jwt : null;
        this.loginService.setLogged(jwt);
        this.spinner = false;
        this.router.navigate(['check-imports-table']);
      });
    }
  }
}
