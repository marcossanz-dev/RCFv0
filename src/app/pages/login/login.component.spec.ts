import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';
import { MatButtonModule } from '@angular/material/button';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatIconModule } from '@angular/material/icon';
import { MatInputModule } from '@angular/material/input';
import { MatRadioModule } from '@angular/material/radio';
import { MatSortModule } from '@angular/material/sort';
import { MatTableModule } from '@angular/material/table';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { ResizableModule } from 'angular-resizable-element';
import { AppComponent } from 'src/app/app.component';
import { FilterComponent } from '../components/filter/filter.component';
import { TableComponent } from '../components/table/table.component';
import { HomeComponent } from '../home/home.component';
import { SharedModule } from '../shared.module';

import { LoginComponent } from './login.component';

describe('LoginComponent', () => {
  let component: LoginComponent;
  let fixture: ComponentFixture<LoginComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [AppComponent,
      ],
      imports: [
        SharedModule
      ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(LoginComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
