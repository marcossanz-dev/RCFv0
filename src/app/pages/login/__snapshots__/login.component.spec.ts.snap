// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component snapshots test should be matched with initial template 1`] = `
<app-login
  loginForm={[Function FormGroup]}
  loginService={[Function LoginService]}
  router={[Function Router]}
  spinner="false"
>
  <h1>
    Login View
  </h1><div
    class="login__form"
  >
    <form
      novalidate=""
    >
      <div
        class="form-input"
      >
        <label
          class="label"
        >
          Username
        </label>
        <input
          class="general-input"
          formcontrolname="username"
          type="text"
        />
        
      </div>
      <div
        class="form-input"
      >
        <label
          class="label"
        >
          Password
        </label>
        <input
          class="general-input"
          formcontrolname="password"
          type="password"
        />
        
      </div>
      <div
        class="form-input"
      >
        <button
          class="submit-button"
        >
           Enter 
        </button>
      </div>
    </form>
  </div>
</app-login>
`;
